### PR TITLE
Ginkgo test for karmor recommend

### PR DESCRIPTION
**Pull request by vishnusomank**
- added basic test for `karmor recommend` using ginkgo framework

Signed-off-by: Vishnu Soman <vishnu@accuknox.com>

### CODE REVIEW

File: tests/recommend/recommend_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	AfterEach(func() {
+		os.RemoveAll(testOptions.OutDir)
+		testOptions.Images = []string{}
+		testOptions.Namespace = ""
+	})
+
+	Describe("Command", func() {
+		Context("when called with image name ubuntu", func() {
+
+			It("should fetch the ubuntu:latest image and create a directory ubuntu-latest under out folder", func() {
```
Comment (Ankurk99) [inline tests/recommend/recommend_test.go:41]: 'out' folder

File: tests/recommend/res/wordpress-mysql-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    spec:
+      containers:
+      - name: mysql
+        image: mysql:5.6
+        env:
+        - name: MYSQL_ROOT_PASSWORD
+          value: root-password
+        ports:
+        - name: mysql
+          containerPort: 3306
```
Comment (Ankurk99) [inline tests/recommend/res/wordpress-mysql-deployment.yaml:96]: pls remove the extra line at eof

File: tests/Makefile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 Authors of KubeArmor
+
+.PHONY: recommend
+recommend:
+	@go mod tidy
+	@ginkgo run -r recommend --timeout=30m
```
Comment (Ankurk99) [inline tests/Makefile:7]: pls remove the extra line at eof

**Review by Ankurk99 — COMMENTED** [review]
Mostly LGTM, just few minor nits suggestions

**Comment by Ankurk99** [comment]
Also can you please check why the CI tests are failing?

**Comment by vishnusomank** [comment]
> Also can you please check why the CI tests are failing?

Thanks for the review @Ankurk99. I'll check why CI failed.

**Review by Ankurk99 — APPROVED** [review]
LGTM!

**Review by nyrahul — APPROVED** [review]
we need to add validation for verifying if the rules that are generated are what is expected. Currently it just checks for the creation of policy files.

**Comment by Ankurk99** [comment]
@vishnusomank please create an issue for the tests to verify generated policies.

**Pull request merged**